subsys/spm/spm.c

```diff
@@ -288,180 +288,175 @@ static int spm_config_unlocked_peripheral(uint8_t id, bool dma_present)
 	return config_peripheral(id, dma_present, false);
 }
 
-static void spm_dppi_configure(uint32_t mask)
-{
-	NRF_SPU->DPPI[0].PERM = mask;
-}
-
 static void spm_config_peripherals(void)
 {
 	struct periph_cfg {
 #ifndef CONFIG_SPM_BOOT_SILENTLY
 		char *name;
 #endif
 		uint8_t id;
-		uint8_t nonsecure;
+		uint8_t secure;
 	};
 
 	/* - All user peripherals are allocated to the Non-Secure domain.
 	 * - All GPIOs are allocated to the Non-Secure domain.
 	 */
 	static const struct periph_cfg periph[] = {
 #ifdef NRF_P0
-		PERIPH("NRF_P0", NRF_P0, CONFIG_SPM_NRF_P0_NS),
+		PERIPH("NRF_P0", NRF_P0, CONFIG_NRF_GPIO0_SECURE),
 #endif
 #ifdef NRF_CLOCK
-		PERIPH("NRF_CLOCK", NRF_CLOCK, CONFIG_SPM_NRF_CLOCK_NS),
+		PERIPH("NRF_CLOCK", NRF_CLOCK, CONFIG_NRF_CLOCK_SECURE),
```

Comment:
Its almost hard to look at this not being pulled out into macros

```
/* Or whatever the syntax would be */
#define CFG_PERIPH(periph) PERIPH("NRF_#periph#", NRF_#periph#, CONFIG_NRF_#periph#_SECURE),
/* ... */
CFG_PERIPH(CLOCK)
CFG_PERIPH(RTC0)
/* ... */
```

But I guess we won't invest much in spm.c anymore, so don't care about this comment.

Reply:
Not worth spending time on, and there are complications where NRFX and MDK don't have the same name, e.g. USBREG - USBREGULATORS